typings.d.ts

```diff
@@ -76,13 +76,13 @@ declare module 'redux-undo' {
   }
 
   interface Undoable {
-    <State, A extends Action = AnyAction>(reducer: Reducer<State, A>, options?: UndoableOptions): Reducer<StateWithHistory<State>>;
+    <S = any, A extends Action = AnyAction>(reducer: Reducer<S, A>, options?: UndoableOptions<S, A>): Reducer<StateWithHistory<S>>;
   }
 
-
-  type IncludeAction = (actions: string | string[]) => FilterFunction;
+  type IncludeAction = <S = any, A extends Action = AnyAction>(actions: A['type'] | A['type'][]) => FilterFunction<S, A>;
   type ExcludeAction = IncludeAction;
-  type GroupByActionTypes = (actions: string | string[]) => GroupByFunction;
+  type CombineFilterFunctions = <S = any, A extends Action = AnyAction>(...filters: FilterFunction<S, A>[]) => FilterFunction<S, A>;
```

Comment:
I added this in order to keep `combineFilters` generic types optional but still enforce types.

Otherwise, if I used the above `export type CombineFilters<S = any, A extends Action = AnyAction>`, the arguments to `combineFilters()` would become loosely-typed.

Please let me know if there's a better way to resolve this.

Reply:
Gotcha.

In this case, I think it would be best to set `CombineFilters` to what you have on that line and remove `CombineFilterFunctions`.

```diff
- export type CombineFilters<S = any, A extends Action = AnyAction> = (...filters: FilterFunction<S, A>[]) => FilterFunction<S, A>;
+ export type CombineFilters = <S = any, A extends Action = AnyAction>(...filters: FilterFunction<S, A>[]) => FilterFunction<S, A>;
```

If a user is basing their types on `CombineFilters`, they should be using the generics along with it.

It should be good besides that.

Reply:
Ah, that makes sense. Thanks!